File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    bin_convert, zero_removed = 0, 0
    while s != '1':
        num = s.count('1')
        zero_removed += len(s) - num
        bin_convert += 1
        s = bin(num)[2:]    # 0b~ 형태이므로 슬라이싱으로 숫자만 잘라줌
    return [bin_convert, zero_removed]

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let [num, bin_convert, zero_removed] = [0, 0, 0];
    while (s !== '1') {
        num = s.split('').reduce((cnt, v) => v === '1' ? cnt + 1 : cnt, 0);
        zero_removed += s.length - num;
        bin_convert ++;
        s = num.toString(2);
    }
    return [bin_convert, zero_removed];
}